File: Semester 3/AD/Experiment 5/q01.py
// Program 1: Check whether a list is a MIN-Priority Queue

import java.util.Scanner;

public class Program1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter size:");
        int[] arr = new int[sc.nextInt()];

        System.out.println("Enter elements:");
        for (int i = 0; i < arr.length; i++) {
            arr[i] = sc.nextInt();
        }

        boolean isMinHeap = true;

        for (int i = 0; i <= (arr.length - 2) / 2; i++) {
            int left = 2 * i + 1;
            int right = 2 * i + 2;

            if (left < arr.length && arr[i] > arr[left]) {
                isMinHeap = false;
                break;
            }
            if (right < arr.length && arr[i] > arr[right]) {
                isMinHeap = false;
                break;
            }
        }

        if (isMinHeap)
            System.out.println("YES");
        else
            System.out.println("NO");
    }
}

/*
Input:
1 3 2 7 4 6 5 8
Output:
YES
*/

File: Semester 1/Assignment 2/Q01.java
package Assignment2;
import java.util.Scanner;
public class Q01{
	public static void main(String[] args) {
		System.out.print("Enter Temperature in Fahrenheit  ");
		Scanner sc =new Scanner(System.in);
		double f= sc.nextDouble();
		double c=(f-32)*5.0/9;
		System.out.println(f+" Fahrenheit is "+c+" Celsius ");
		sc.close();
	}

}
// output:
// Enter Temperature in Fahrenheit  98.6 Fahrenheit is 37.0 Celsius 
